### PR TITLE
Plenty 472 migrate response to json with variants without api

### PR DESCRIPTION
**Pull request by rimvydas-hashtages**

### CODE REVIEW

File: tests/MockResponses/someResultsWithFilters.json
Comment (TobiasGraml11) [inline tests/MockResponses/someResultsWithFilters.json:1]: - Guess we can remove the corresponding XML file?
- What about noResults.xml, is it not used or why don't we need a JSON version of it?

File: src/Api/Response/ResponseParser.php [hunk 1]
```diff
@@ -51,171 +53,121 @@ public function parse(HttpRequest $request, $responseData): Response
...
 
-                foreach ($product->properties->property as $property) {
-                    $productData['properties'][$property['name']->__toString()] = $property->__toString();
-                }
+        $queryStringType = null;
 
-                $products[] = $productData;
-            }
+        if($improvedQuery){
+            $queryStringType = 'improved';
```
Comment (TobiasGraml11) [inline src/Api/Response/ResponseParser.php]: 1. The response class has constants for the query types
2. Do we never need the `didYouMean` type?

File: src/Api/Response/Parser/FiltersParser.php [hunk 2]
```diff
@@ -107,45 +105,49 @@ public function parseForWidgets($data): array
...
-    public function parseFilterItem($filterType, &$filterItem, $data, $index)
+    public function parseFilterItem($filterType, &$filterItem, array $data, $index)
     {
         if (!empty($data)) {
             $filterItem['items'] = [];
-            $filterItem['name'] = $data->name->__toString();
-            $filterItem['position'] = $index;
-            $filterItem['count'] = $data->frequency->__toString();
+            $filterItem['name'] = $data['value'];
+            $filterItem['position'] = "item";
```
Comment (TobiasGraml11) [inline src/Api/Response/Parser/FiltersParser.php]: Why did you change the `position` value?

File: src/Api/Response/Parser/FiltersParser.php [hunk 3]
```diff
@@ -107,45 +105,49 @@ public function parseForWidgets($data): array
...
+            $filterItem['count'] = array_key_exists('frequency', $data) ? (string) $data['frequency'] : '';
```
Comment (TobiasGraml11) [inline src/Api/Response/Parser/FiltersParser.php]: `$data['frequency] ?? ''` should be enough. The operator has the key check "included". Can probably be applied in some other places as well.

**Review by TobiasGraml11 — CHANGES_REQUESTED** [review]

File: src/Api/Response/Parser/FiltersParser.php (hunk 2 above)
Comment (rimvydas-hashtages) [reply src/Api/Response/Parser/FiltersParser.php]: @TobiasGraml11 why I named it 'item' everywhere is because in the develop branch every filter item basically has 'item' in it as the position
image

Comment (TobiasGraml11) [reply src/Api/Response/Parser/FiltersParser.php]: Does the `$index` variable contain "item"? If so, I think it still makes sense to keep it like it is and don't hardcode it there.

Comment (rimvydas-hashtages) [reply src/Api/Response/Parser/FiltersParser.php]: No, it contains numbers. Guess in xml it was always 'item' cause the xml is something like this:
`<items>
   <item>
      <id>....
    </item>
 </items>`

Comment (TobiasGraml11) [reply src/Api/Response/Parser/FiltersParser.php]: Interesting, please still stick with `$filterItem['position'] = $index;`. I don't think it's used anywhere, but having the numeric values makes more sense.

Comment (rimvydas-hashtages) [reply src/Api/Response/Parser/FiltersParser.php]: @TobiasGraml11 check it again now

File: src/Api/Response/ResponseParser.php (hunk 1 above)
Comment (TobiasGraml11) [reply src/Api/Response/ResponseParser.php]: @rimvydas-hashtages Only one more thing: Please let me know why we never need to set the `didYouMean` type.

Comment (rimvydas-hashtages) [reply src/Api/Response/ResponseParser.php]: Because old code always overrides the type to 'did-you-mean' whenever the 'didYouMeanQuery' is set:
image
image

**Review by TobiasGraml11 — APPROVED** [review]
